packages/xstate-react/src/useInterpret.ts

```diff
@@ -82,6 +82,12 @@ export function useIdleInterpreter(
     Object.assign(service.machine.options.delays!, delays);
   }, [actions, guards, activities, services, delays]);
 
+  useIsomorphicLayoutEffect(() => {
+    if (options.subscribe) {
+      return service.subscribe(options.subscribe).unsubscribe;
+    }
+  }, []);
```

Comment:
This has been applied in the internal `useIdleInterpret` and now `useInterpret` has two ways t do the same thing, as it already supports such subscribers as its third parameter, see here:
https://github.com/statelyai/xstate/blob/cd79700880890675af7d0e96516a2622630e2905/packages/xstate-react/src/useInterpret.ts#L131-L141

Reply:
Relates to https://github.com/statelyai/xstate/pull/3688#discussion_r1035087432